### PR TITLE
[bug-fix] remove reference to json_array, as it's not a doctrine type…

### PR DESCRIPTION
**Pull request by ghost**
[bugfix]

Doctrine JSON_ARRAY type's is deprecated and not used anymore in doctrine 3.*

here is the bug I encountered 
```
!!  14:54:51 CRITICAL  [php] Uncaught Error: Undefined constant Doctrine\DBAL\Types\Types::JSON_ARRAY ["exception" => Error { …}]
!!  
!!  In DoctrineExtractor.php line 283:
!!                                                              
!!    Undefined constant Doctrine\DBAL\Types\Types::JSON_ARRAY  
```

How to reproduce:
Here is how I encountered this bug. 

my composer.json has :  
have a doctrine with 3.* version
symfony with 5.*

### CODE REVIEW

File: src/Symfony/Bridge/Doctrine/PropertyInfo/DoctrineExtractor.php [hunk 1]
```diff
@@ -167,8 +167,6 @@ public function getTypes(string $class, string $property, array $context = [])
                 case Type::BUILTIN_TYPE_ARRAY:
                     switch ($typeOfField) {
                         case Types::ARRAY:
-                        case 'json_array':
-                            return [new Type(Type::BUILTIN_TYPE_ARRAY, $nullable, null, true)];
```
Comment (stof) [inline src/Symfony/Bridge/Doctrine/PropertyInfo/DoctrineExtractor.php:171]: this removes the return for `case Types::ARRAY:`, which is totally wrong

**Review by stof — CHANGES_REQUESTED** [review]
The min supported version of doctrine/dbal in Symfony 5.4 is 2.13.1, not 3.0. So we cannot remove support for the `json_array` type yet.

**Comment by stof** [comment]
And the code you removed does not use the `Doctrine\DBAL\Types\Types::JSON_ARRAY` constant, so it cannot be the source of your error message.

**Comment by xabbuh** [comment]
In #43975 there was a report about the constant being used. But that was for 5.2 which is not supported anymore.

**Pull request closed without merging**